src/Util/FilenameUtils.php

```diff
@@ -19,12 +23,26 @@ private function __construct()
      *
      * @return array An array of basename and extension
      */
-    public static function spitNameByExtension(string $filename): array
+    public static function splitNameByExtension(string $filename): array
     {
-        if (false === $pos = \strrpos($filename, '.')) {
+        if (false === $pos = strrpos($filename, '.')) {
             return [$filename, ''];
         }
 
-        return [\substr($filename, 0, $pos), \substr($filename, $pos + 1)];
+        return [substr($filename, 0, $pos), substr($filename, $pos + 1)];
+    }
+
+    /**
+     * Splits filename for array of basename and extension.
+     *
+     * @return array An array of basename and extension
+     *
+     * @deprecated, use splitNameByExtension() instead
+     */
+    public static function spitNameByExtension(string $filename): array
+    {
+        trigger_deprecation('vich/uploader-bundle', '2.0.2', '"%s()" is deprecated, use "splitNameByExtension()" instead.', __METHOD__);
```

Comment:
We don't add deprecation in patch versions. Please change this with `2.1.0`

Reply:
Changed to 2.1.0 as requested.

Reply:
On second thought, I realized that we don't need to deprecate this method, since the class is marked as "internal".
By the way, the class itself looks a bit overkill, since it's used only in our `Transliterator` (that's internal too).
So, please change this PR (or open a new one, if you prefer) to remove the class and move the method as a private method of the `Transliterator` class

Reply:
Don't you want to give users the chance to remove this function call in case they just ignored that it is marked as "internal"? I could then change the comment to state that this function will be removed in the next major release?

Reply:
I don't want to wait for the next major release. We'll add a note about the breaking change in the UPGRADE file: if someone is using the internal class, it's up to them to use caution with upgrades.

Reply:
I moved splitNameByExtension to /src/Util/Transliterator.php updated the UPGRADE.md, removed /src/Util/FilenameUtils.php and /tests/Util/FilenameUtilsTest.php.